pkg/controller/update_cluster.go

```diff
@@ -105,6 +109,43 @@ func (c *M3DBController) createNamespaces(cluster *myspec.M3DBCluster, registry
 	return nil
 }
 
+// readyNamespaces will attempt to mark all namespaces in the initializing state as ready.
+func (c *M3DBController) readyNamespaces(cluster *myspec.M3DBCluster, registry *dbns.Registry) error {
+	// NB(nate): Marking namespaces ready (without force) involves checking dbnodes for namespace
+	// availability. External coordinators do not have connections to dbnodes so don't attempt this
+	// step for now.
+	if cluster.Spec.ExternalCoordinator != nil {
+		c.logger.Debug("using an external coordinator. will not attempt to ready namespaces")
+		return nil
+	}
+
+	toReady := namespacesToReady(registry, cluster.Spec.Namespaces)
+	for _, ns := range toReady {
+		req := &admin.NamespaceReadyRequest{
+			Name: ns.Name,
+		}
+		err := c.adminClient.namespaceClientForCluster(cluster).Ready(req)
+		cause := pkgerrors.Cause(err)
+		// NB(nate): Due to bug in coordinator API routing logic, missing routes may
+		// also be returned as 405s (i.e. method not allowed). So check for that in addition
+		// to 404s.
+		if cause == m3admin.ErrNotFound || cause == m3admin.ErrMethodNotAllowed {
```

Comment:
This was a fun little discovery during testing. Depending on the route, misses will either return a `404` or a `405`. Almost certainly a bug in our routing logic, but since these coordinators will be out there in the wild, we need to account for it.

(cc @schallert since this is a small update after the LGTM)